cmd/relay/bgs/validator.go

```diff
@@ -305,46 +368,75 @@ func (val *Validator) VerifyCommitMessage(ctx context.Context, host *models.PDS,
 func (val *Validator) HandleSync(ctx context.Context, host *models.PDS, msg *atproto.SyncSubscribeRepos_Sync) (newRoot *cid.Cid, err error) {
 	hostname := host.Host
 	hasWarning := false
+	logger := slog.Default().With("host", hostname, "did", msg.Did, "rev", msg.Rev, "seq", msg.Seq, "time", msg.Time)
 
 	did, err := syntax.ParseDID(msg.Did)
 	if err != nil {
 		syncVerifyErrors.WithLabelValues(hostname, "did").Inc()
-		return nil, err
+		if !val.Sync11ErrorsAreWarnings {
+			return nil, err
+		} else {
+			logger.Warn("invalid did", "err", err)
+		}
 	}
 	rev, err := syntax.ParseTID(msg.Rev)
 	if err != nil {
 		syncVerifyErrors.WithLabelValues(hostname, "tid").Inc()
-		return nil, err
+		if !val.Sync11ErrorsAreWarnings {
+			return nil, err
+		} else {
+			logger.Warn("invalid rev", "err", err)
+		}
 	}
 	if rev.Time().After(time.Now().Add(val.maxRevFuture)) {
 		syncVerifyErrors.WithLabelValues(hostname, "revf").Inc()
-		return nil, val.ErrRevTooFarFuture
+		if !val.Sync11ErrorsAreWarnings {
+			return nil, val.ErrRevTooFarFuture
+		} else {
+			logger.Warn("invalid rev too far future", "now", time.Now(), "rev", rev.Time())
+		}
 	}
 	_, err = syntax.ParseDatetime(msg.Time)
 	if err != nil {
 		syncVerifyErrors.WithLabelValues(hostname, "time").Inc()
-		return nil, err
+		if !val.Sync11ErrorsAreWarnings {
+			return nil, err
+		} else {
+			logger.Warn("invalid time", "err", err)
+		}
 	}
 
 	commit, err := atrepo.LoadCARCommit(ctx, bytes.NewReader([]byte(msg.Blocks)))
 	if err != nil {
 		commitVerifyErrors.WithLabelValues(hostname, "car").Inc()
-		return nil, err
+		if !val.Sync11ErrorsAreWarnings {
+			return nil, err
+		} else {
+			logger.Warn("invalid car", "err", err)
+		}
 	}
 
 	if commit.Rev != rev.String() {
```

Comment:
If there was an error in loading the CAR, I would expect `commit` to be `nil`.  If that's the case, falling through on the "invalid car" warn would cause a nil pointer exception here.

Reply:
fixed (and in commit handler)